velox/exec/trace/QueryDataReader.cpp

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "velox/exec/trace/QueryDataReader.h"
+
+#include "velox/common/file/File.h"
+#include "velox/connectors/hive/HiveDataSink.h"
+#include "velox/connectors/hive/TableHandle.h"
+#include "velox/exec/TableWriter.h"
+#include "velox/exec/trace/QueryTraceTraits.h"
+
+namespace facebook::velox::exec::trace {
+
+QueryDataReader::QueryDataReader(std::string path) : path_(std::move(path)) {
+  if (!isRegisteredVectorSerde()) {
+    serializer::presto::PrestoVectorSerde::registerVectorSerde();
+  }
+
+  const auto fs = filesystems::getFileSystem(path_, nullptr);
```

Comment:
Make fs a class member? And it seems that both dataType_ and dataStream_ could be const members?